frontend/packages/rhoas-plugin/src/topology/components/KafkaNode.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import {
+  observer,
+  Node,
+  useDndDrop,
+  WithContextMenuProps,
+  WithCreateConnectorProps,
+  WithDragNodeProps,
+  WithSelectionProps,
+} from '@patternfly/react-topology';
+import { calculateRadius } from '@console/shared';
+import { RootState } from '@console/internal/redux';
+import { getServiceBindingStatus } from '@console/topology/src/utils';
+import { obsDropTargetSpec } from '@console/topology/src/operators/components/OperatorBackedService';
+import { KafkaConnectionModel } from '../../models';
+import { kafkaIcon } from '../../const';
+import TrapezoidBaseNode from './TrapezoidBaseNode';
+
+import './KafkaNode.scss';
+
+interface StateProps {
+  serviceBinding: boolean;
+}
+
+type KafkaNodeProps = {
+  element: Node;
+} & WithSelectionProps &
+  WithDragNodeProps &
+  WithContextMenuProps &
+  WithCreateConnectorProps &
+  StateProps;
+
+const KafkaNode: React.FC<KafkaNodeProps> = ({
+  element,
+  selected,
+  onSelect,
+  serviceBinding,
+  ...props
+}) => {
+  const { width, height } = element.getBounds();
+  const size = Math.min(width, height);
+  const iconRadius = Math.min(width, height) * 0.25;
+  const { radius } = calculateRadius(size);
+  const spec = React.useMemo(() => obsDropTargetSpec(serviceBinding), [serviceBinding]);
+  const [dndDropProps, dndDropRef] = useDndDrop(spec, { element, ...props });
```

Comment:
For this I would need help of @jeff-phillips-18 . Can we have all props passed?

Reply:
The additional props should only be passed if the the `spec` requires them.
In this case it looks like the spec only requires `element` as an additional prop. 

We really should look at providing some common spec or simply writing one for this particular use case so that we don't reuse a spec that was meant for the operator backed service nodes. cc @jeff-phillips-18

Reply:
Agreed. Reused this for now since we will be adding extensions to be able to create edges.

Reply:
Are there any actions needed from us here?

Reply:
I thought you only need to provide `element`. But I suppose it's no big deal since the rest are simply ignored and unused.